contracts/RecurringTransfersModule.sol

```diff
@@ -0,0 +1,225 @@
+        require(msg.sender == recurringTransfer.delegate || OwnerManager(manager).isOwner(msg.sender), "Method can only be called by a delegate or owner");
+        require(isPastMonth(recurringTransfer.lastTransferTime), "Transfer has already been executed this month");
+        require(isOnDayAndBetweenHours(recurringTransfer.transferDay, recurringTransfer.transferHourStart, recurringTransfer.transferHourEnd), "Transfer request not within valid timeframe");
+
+        uint256 transferAmount = getAdjustedTransferAmount(recurringTransfer.token, recurringTransfer.rate, recurringTransfer.amount);
+
+        if (recurringTransfer.token == 0) {
+            require(manager.execTransactionFromModule(receiver, transferAmount, "", Enum.Operation.Call), "Could not execute Ether transfer");
+        } else {
+            bytes memory data = abi.encodeWithSignature("transfer(address,uint256)", receiver, transferAmount);
+            require(manager.execTransactionFromModule(recurringTransfer.token, 0, data, Enum.Operation.Call), "Could not execute token transfer");
+        }
+
+        recurringTransfers[receiver].lastTransferTime = now;
+
+        // We transfer the calculated tx costs to the tx.origin to avoid sending it to intermediate contracts that have made calls
+        if (gasPrice > 0) {
+            require(gasPrice <= gasPriceLimits[gasToken], "Gas price is too high");
+            require(dataGas <= dataGasLimits[gasToken], "Data gas is too high");
+            handlePayment(startGas, dataGas, gasPrice, gasToken, refundReceiver);
+        }
+    }
+
+    function isOnDayAndBetweenHours(uint8 day, uint8 hourStart, uint8 hourEnd)
+        internal view returns (bool)
+    {
+        return dateTime.getDay(now) == day &&
+        dateTime.getHour(now) >= hourStart &&
+        dateTime.getHour(now) < hourEnd;
+    }
+
+    function isPastMonth(uint256 previousTime)
+        internal view returns (bool)
+    {
+        return dateTime.getYear(now) > dateTime.getYear(previousTime) ||
+        dateTime.getMonth(now) > dateTime.getMonth(previousTime);
+    }
+
+    // Adjust amount for the transfer rate if it exists
+    // For example:
+    // say GNO = 1/10 ETH and DAI = 1/200 ETH
+    // token = GNO address
+    // rate = DAI address
+    // amount = 1000
+    // In other words, we want to pay the receiver $1000 worth of GNO token.
+    // Given the rates above, we will pay (1 * 10 * 1000) / (200 * 1) = 50 GNO tokens.
+    function getAdjustedTransferAmount(address token, address rate, uint256 amount)
+        internal view returns (uint)
+    {
+        // transfer does not need to be adjusted since no rate is given
+        if(rate == 0) {
+            return amount;
+        }
+
+        uint256 tokenPriceNum = 1;
+        uint256 tokenPriceDen = 1;
+        // they are transfering ETH
+        if(token != 0) {
+            (tokenPriceNum, tokenPriceDen) = dutchExchange.getPriceOfTokenInLastAuction(token);
```

Comment:
@anxolin could you check the dutchx logic here

Reply:
The logic looks ok. 

The only 2 small inputs I can give are:
- To get the new amount for GNO-DAI it's getting the prices of GNO-ETH and DAI-ETH. I find this solution simple and good enough. But note that the GNO-DAI market may also be created so `getPriceInPastAuction(token, rate, lastAuction)` could get that specific price
- The math operations, should be protected with a safe math lib, right?
- The function is internal, but I find that a user interface might find useful to show the user the amount they will pay in any token before executing the transaction. So I would make it public.

Reply:
I will make the following changes:
- attempt to get the value between two tokens before calculating based on the eth value
- use safe math
- make the method public
- create an interface and implement it

Reply:
Thanks for the feedback @anxolin